### PR TITLE
case-lib: fix sof-test.lock refer problem

### PR DESCRIPTION
**Pull request by Bin-QA**
1. Add pid detect to avoid perpare case
abnormal exit without lock clean
2. follow Liam suggest change from 'skip' => 'Incomplete'
Add new exit status:3 for 'Incomplete'

Signed-off-by: Wu, BinX <binx.wu@intel.com>

### CODE REVIEW

File: case-lib/lib.sh [hunk 1]
```diff
@@ -23,7 +23,8 @@ if [ ! -f "$SOF_LOCK" ];then # lock is not exist
     echo $$ > /tmp/sof-test.lock # write self pid into lock file
 elif [ ! "$(alias |grep -i 'Sub-Test')" ]; then # not the sub test-case
     echo "Find $SOF_LOCK already exist: $(ps -p $(cat $SOF_LOCK))"
-    exit 2 # now skip to run the test-case
+    # exit 2 # now skip to run the test-case
+    exit 1 # mark test-case is failed
```
Comment (stripes416) [inline case-lib/lib.sh]:
```suggestion
    exit 1 # mark test-case as failed
```
Additionally, you've got typos in your commit message -: exsit -> exist; as -> is

**Comment by lgirdwood** [comment]
@Bin-QA @xiulipan any CI failure is not a test failure. This is confusing for developers. Can we have the following CI codes.
```
Failed (in red) = Test has failed
N/A (in grey) = test is not applicable for this target or topology
Incomplete (in yellow) = CI cant run this test or CI script had internal error unrelated to test
```
Any other categories, what does skipped mean ?

**Comment by Bin-QA** [comment]
> @Bin-QA @xiulipan any CI failure is not a test failure. This is confusing for developers. Can we have the following CI codes.
> 
> ```
> Failed (in red) = Test has failed
> N/A (in grey) = test is not applicable for this target or topology
> Incomplete (in yellow) = CI cant run this test or CI script had internal error unrelated to test
> ```
> 
> Any other categories, what does skipped mean ?

Normally skip means 'N/A'  
Because test script need to check some environment/ some thing like TPLG pipeline  
when the environment is not enough to run the test case  
we will mark it as 'skip'  

@xiulipan 
here refer our test-report we ignore 'skip' as 'pass', so we want to trace those 'skip'

**Comment by lgirdwood** [comment]
@Bin-QA so lets change "skip" to "N/A".

**Comment by stripes416** [comment]
i'm sorry --- I missed a typo in your commit message --- perpare --> prepare

**Comment by xiulipan** [comment]
@lgirdwood We are having a full stack clean up for this naming update. It is really needed to identify the issue from CI and DUT configuration. Will send PR later to also update the 01.org output.

**Comment by lgirdwood** [comment]
@xiulipan great, thanks - we should also put this information on sof-docs so people can understand the CI results better.

**Comment by xiulipan** [comment]
@lgirdwood Good idea, I will create an issue to track this.

**Pull request merged**